src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import com.ctre.phoenix.sensors.CANCoder;
+import com.ctre.phoenix.sensors.CANCoderConfiguration;
+import com.ctre.phoenix.sensors.SensorTimeBase;
+import com.revrobotics.AbsoluteEncoder;
+import com.revrobotics.SparkMaxPIDController;
+import com.revrobotics.CANSparkMax.ControlType;
+import com.revrobotics.SparkMaxAbsoluteEncoder;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.SubsystemChecker;
+import frc.robot.SubsystemChecker.SubsystemType;
+import frc.robot.config.Config;
+
+public class ArmSubsystem extends SubsystemBase {
+
+  private static final MotorType motorType = MotorType.kBrushless;
+  private static final SparkMaxAbsoluteEncoder.Type encAbsType = SparkMaxAbsoluteEncoder.Type.kDutyCycle;
+  
+  public CANCoder m_absoluteTopArmEncoder;
+  public CANCoder m_absoluteBottomArmEncoder;
+
+
+  private static ArmSubsystem instance = null;
+
+
+  public final CANSparkMax m_topArm;
+  public final CANSparkMax m_bottomArm;
+  public SparkMaxPIDController m_pidControllerTopArm;
+  public SparkMaxPIDController m_pidControllerBottomArm;
+  public double kP, kI, kD, kIz, kFF, kMaxOutput, kMinOutput;
+
+  /** Creates a new ArmSubsystem. */
+  public ArmSubsystem() {
+    m_topArm = new CANSparkMax(Config.CANID.TOP_ARM_SPARK_CAN_ID, motorType);
+    m_bottomArm = new CANSparkMax(Config.CANID.BOTTOM_ARM_SPARK_CAN_ID, motorType);
+    CANCoderConfiguration config = new CANCoderConfiguration();
+    m_topArm.restoreFactoryDefaults();
+    m_bottomArm.restoreFactoryDefaults();
+
+    // set units of the CANCoder to radians, with velocity being radians per second
+    config.unitString = "rad";
+    config.sensorTimeBase = SensorTimeBase.PerSecond;
+
+    m_absoluteTopArmEncoder = new CANCoder(16);
+    m_absoluteBottomArmEncoder = new CANCoder(17);
+
+    m_absoluteTopArmEncoder.configAllSettings(config);
+    m_absoluteBottomArmEncoder.configAllSettings(config);
+
+    m_pidControllerTopArm = m_topArm.getPIDController();
+    m_pidControllerBottomArm = m_bottomArm.getPIDController();
+
+    // PID coefficients (probably need to change values and put the values in config)
+    kP = 0.02;
```

Comment:
Is there a reason to set these values inside the constructor instead of when you declare them at the top? They kind of feel like magic numbers, which are spooky. (magic numbers are numbers that seemingly come from no where, with no explanation as to what they are or where they came from. They make code pretty impossible to follow. If you're not planning on extracting them and putting them in the config (which I recommend), then I suggest defining and setting these are the top, so it's easy to understand where they are defined, should they need to be changed.